llvm/lib/Target/DirectX/DXILDataScalarization.cpp

```diff
@@ -40,9 +41,10 @@ static bool findAndReplaceVectors(Module &M);
 class DataScalarizerVisitor : public InstVisitor<DataScalarizerVisitor, bool> {
 public:
   DataScalarizerVisitor() : GlobalMap() {}
-  bool visit(Instruction &I);
+  bool visit(Function &F);
```

Comment:
nit: It seems rather odd to make the `visit` accept a Function arg instead of an Instruction, while the other `visit*` functions still take a (subclass of) Instruction. It doesn't really fit the usual visitor pattern.